schnorr_musig.h

```diff
@@ -0,0 +1,114 @@
+  NONCE_COMMITMENTS_AND_PARTICIPANTS_NOT_MATCH,
+  SIGNATURE_SHARE_AND_PARTICIPANTS_NOT_MATCH,
+  COMMITMENT_IS_NOT_IN_CORRECT_SUBGROUP,
+  INVALID_COMMITMENT,
+  INVALID_PUBLIC_KEY,
+  INVALID_PARTICIPANT_POSITION,
+  AGGREGATED_NONCE_COMMITMENT_NOT_COMPUTED,
+  CHALLENGE_NOT_GENERATED,
+  INVALID_SIGNATURE_SHARE,
+  INVALID_SEED,
+} MusigRes;
+
+typedef struct MusigBN256Signer MusigBN256Signer;
+
+typedef struct AggregatedPublicKey {
+  uint8_t data[STANDARD_ENCODING_LENGTH];
+} AggregatedPublicKey;
+
+typedef struct MusigSigner {
+  struct MusigBN256Signer *inner;
+} MusigSigner;
+
+typedef struct Precommitment {
+  uint8_t data[STANDARD_ENCODING_LENGTH];
+} Precommitment;
+
+typedef struct AggregatedCommitment {
+  uint8_t data[STANDARD_ENCODING_LENGTH];
+} AggregatedCommitment;
+
+typedef struct Commitment {
+  uint8_t data[STANDARD_ENCODING_LENGTH];
+} Commitment;
+
+typedef struct Signature {
+  uint8_t data[STANDARD_ENCODING_LENGTH];
+} Signature;
+
+typedef struct AggregatedSignature {
+  uint8_t data[AGG_SIG_ENCODING_LENGTH];
+} AggregatedSignature;
+
+enum MusigRes schnorr_musig_aggregate_pubkeys(const uint8_t *encoded_pubkeys,
+                                              size_t encoded_pubkeys_len,
+                                              struct AggregatedPublicKey *aggregate_pubkeys);
+
+enum MusigRes schnorr_musig_compute_precommitment(struct MusigSigner *signer,
+                                                  const uint32_t *seed,
+                                                  size_t seed_len,
+                                                  struct Precommitment *precommitment);
+
+void schnorr_musig_delete_signer(struct MusigSigner*);
+
+struct MusigSigner *schnorr_musig_new_signer(const uint8_t *encoded_pubkeys,
+                                             size_t encoded_pubkeys_len,
+                                             size_t position);
+
+enum MusigRes schnorr_musig_receive_commitments(struct MusigSigner *signer,
+                                                const uint8_t *input,
+                                                size_t input_len,
+                                                struct AggregatedCommitment *agg_commitment);
+
+enum MusigRes schnorr_musig_receive_precommitments(struct MusigSigner *signer,
+                                                   const uint8_t *input,
+                                                   size_t input_len,
+                                                   struct Commitment *commitment);
+
+enum MusigRes schnorr_musig_sign(struct MusigSigner *signer,
+                                 const uint8_t *private_key,
+                                 size_t private_key_len,
+                                 const uint8_t *message,
+                                 size_t message_len,
+                                 struct Signature *signature);
+
+enum MusigRes schnorr_musig_signature_shares(struct MusigSigner *signer,
```

Comment:
This was originally named `receive_signature_shares` though, not sure if the long name worth it

Reply:
Oh, this is my fault. I think the better way to leave names the equals to rs.